neurovault/apps/users/views.py

```diff
@@ -80,155 +82,15 @@ def password_change_done(request):
-        form.instance.user = self.request.user
-        messages.success(
-            self.request, "The application has been successfully registered."
-        )
-
-        return super(ApplicationRegistration, self).form_valid(form)
-
-
-class ApplicationUpdate(ApplicationOwnerIsUserMixin, UpdateView):
-
-    """
-    View used to update an application owned by the request.user
-    """
-
-    context_object_name = "application"
-    template_name = "oauth2_provider/application_form.html"
-
-    # Reset the inherited fields attribute and use a form_class instead
-    fields = None
-    form_class = ApplicationEditForm
-
-    def get_success_url(self):
-        return reverse("users:developerapps_list")
-
-    def form_valid(self, form):
-        messages.success(self.request, "The application has been successfully updated.")
-        return super(ApplicationUpdate, self).form_valid(form)
-
-
-class ApplicationDelete(ApplicationOwnerIsUserMixin, DeleteView):
-
-    """
-    View used to delete an application owned by the request.user
-    """
-
-    context_object_name = "application"
-    success_url = reverse_lazy("users:developerapps_list")
-    template_name = "oauth2_provider/application_confirm_delete.html"
-    success_message = "The application has been successfully deleted."
-
-    def delete(self, request, *args, **kwargs):
-        messages.success(self.request, self.success_message)
-        return super(ApplicationDelete, self).delete(request, *args, **kwargs)
-
-
-class ConnectionList(LoginRequiredMixin, ListView):
-    template_name = "oauth2_provider/connection_list.html"
-
-    def get_queryset(self):
-        return RefreshToken.objects.filter(user=self.request.user).distinct(
-            "application"
-        )
-
-
-class ConnectionDelete(LoginRequiredMixin, DeleteView):
-    template_name = "oauth2_provider/connection_confirm_delete.html"
-    success_url = reverse_lazy("connection_list")
-    success_message = "The application authorization has been successfully " "revoked."
-
-    def _refresh_token_queryset(self, user, application_id):
-        return RefreshToken.objects.filter(user=user, application_id=application_id)
-
-    def get_object(self):
-        pk = self.kwargs.get(self.pk_url_kwarg)
-
-        refresh_token = self._refresh_token_queryset(self.request.user, pk).first()
-        if not refresh_token:
-            raise Http404("No application connection found.")
-
-        return refresh_token.application
-
-    def delete(self, request, *args, **kwargs):
-        self.object = self.get_object()
-        success_url = self.get_success_url()
-
-        token_list = self._refresh_token_queryset(self.request.user, self.object.id)
-
-        for refresh_token in token_list:
-            refresh_token.revoke()
-
-        messages.success(self.request, self.success_message)
-
-        return HttpResponseRedirect(success_url)
+@login_required
+def view_token(request, regenerate=False):
```

Comment:
Should we allow people to view tokens that were previously generated? Or be like Github that you can only vie when generating?

Reply:
I'll leave them shown for now, can get more sophisticated in the future.